tests/functional/lib/helpers.js

```diff
@@ -160,8 +236,24 @@ define([
    * @returns {promise} rejects if element does not exist
    */
   const testElementExists = thenify(function (selector) {
+    var findError;
     return this.parent
       .findByCssSelector(selector)
+      .then(null, function (err) {
+        // swallow the error until after taking a screenshot.
+        // The error has to be swallowed before a screenshot
+        // can be taken or else takeScreenshot itself returns
+        // the error.
+        findError = err;
+      })
+      .then(function () {
+        if (findError) {
+          return this.parent.then(takeScreenshot())
```

Comment:
Ahha, this nugget makes sense. I could do that, and you are right, it'd be cleaner.